lib/util.js

```diff
@@ -403,6 +403,108 @@ module.exports = {
     };
   },
 
+  /**
+   * Generates object containing objects of tags and their requests
+   * @param {object} spec - openAPI spec object
+   * @returns {object} returns an object containing objects of tags and their requests
+   */
+  /* eslint-disable max-depth */
+  generateFolderObjectFromTags: function(spec) {
+    var globalTags = spec.tags,
+      localTags = [],
+      globalTag = '',
+      paths = spec.paths,
+      currentPathObject = {},
+      operationObject = {},
+      folderObj = {
+        withTags: {},
+        withoutTags: {
+          requests: []
+        }
+      },
+      collectionVariables = {},
+      commonParams = '',
+      summary = '',
+      pathLevelServers = '',
+      name = '',
+      description = '';
+
+    // checking if tags are defined at the root level
+    if (globalTags.length > 0) {
+      for (let tag in globalTags) {
+        if (globalTags.hasOwnProperty(tag)) {
+          globalTag = globalTags[tag];
+          name = globalTag.name;
+          if (globalTag.hasOwnProperty('description')) {
+            description = globalTag.description;
+          }
+          // adding an object in folder object with tag name as key
+          folderObj.withTags[name] = {
+            description: description,
+            requests: []
+          };
+        }
+      }
+    }
+
+    for (let path in paths) {
+      if (paths.hasOwnProperty(path)) {
```

Comment:
we don't need to use `.hasOwnProperty` if it's a simple object we're constructing ourselves

Reply:
That's leading to most of the max-depth problems, I'm guessing

Reply:
because we've set an eslint rule warning for wrapping everything in an if in for-in loop.